app/views/assessment/AssessmentOption.js

```diff
@@ -0,0 +1,181 @@
+
+/**
+ * @typedef { import("./Assessment").SurveyAnswers } SurveyAnswers
+ * @typedef { import("./Assessment").SurveyOption["values"][0] } SurveyOptionValue
+ */
+
+/** @type {React.FunctionComponent<{
+ *   answer?: SurveyAnswers[0][0];
+ *   index: number;
+ *   onSelect: (value: string) => void;
+ *   option: SurveyOptionValue;
+ *   selected: boolean;
+ *   type: Extract<SurveyScreen, 'Checkbox' | 'Date' | 'Radio'>
+ * }>} */
+const AssessmentOption = ({
+  answer,
+  index,
+  onSelect,
+  option,
+  selected,
+  type,
+}) => {
+  // The API doesn't have a defined way to know a specific option is a date,
+  // we just assume the first option is the date picker when type is SCREEN_TYPE_DATE
+  const isDate = type === SCREEN_TYPE_DATE && index === 0;
+  const showIndicator =
+    type === SCREEN_TYPE_CHECKBOX ||
+    type === SCREEN_TYPE_RADIO ||
+    type === SCREEN_TYPE_DATE;
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [date, setDate] = useState(() =>
+    (isDate && selected && answer) ? new Date(answer.value) : null,
+  );
+  const label = useMemo(() => {
+    if (isDate && date)
+      return `${date.getMonth() + 1}/${date.getDate()}/${date.getFullYear()}`;
+    return option.label;
+  }, [date, isDate, option]);
+  useEffect(() => {
+    if (isDate && !selected) {
+      setDate(null);
+      setShowDatePicker(false);
+    }
+  }, [isDate, selected]);
+  return (
+    <Fragment>
+      <TouchableOpacity
+        onPress={() => {
+          if (isDate) {
+            let date = new Date();
+            setDate(date);
+            setShowDatePicker(true);
+            if (isPlatformiOS()) onSelect(String(date));
+            return;
+          }
+          onSelect(option.value);
+        }}>
+        <View style={[styles.container, selected && styles.containerSelected]}>
+          <View style={styles.primary}>
+            {showIndicator && (
+              <View
+                style={[
+                  styles.indicator,
+                  selected && styles.indicatorSelected,
+                ]}>
+                {selected && type === SCREEN_TYPE_CHECKBOX && (
+                  <SvgXml
+                    height={INDICATOR_WIDTH * 0.5}
+                    style={styles.indicatorIcon}
+                    width={INDICATOR_WIDTH * 0.5}
+                    xml={Icon}
+                  />
+                )}
+                {selected &&
+                  (type === SCREEN_TYPE_RADIO || type === SCREEN_TYPE_DATE) && (
+                    <View style={styles.indicatorIconRadio} />
+                  )}
+              </View>
+            )}
+            <Text style={styles.title}>{label}</Text>
+          </View>
+          <View style={styles.content}>
+            {option.description ? (
+              <Text style={styles.description}>{option.description}</Text>
+            ) : null}
+          </View>
+        </View>
+      </TouchableOpacity>
+      {showDatePicker && (
+        <DateTimePicker
```

Comment:
Could you pull this out into `app/components`? Feels generic enough to go in there and then pull into this file. Eventually, I can imagine us wanting to have some custom theming in there, specific language strings, etc.

Reply:
Also - I wasn't able to come across any DateTimePicker elements during my survey.